fast64_internal/oot/oot_utility.py

```diff
@@ -939,3 +939,41 @@ def getNewPath(type: str, isClosedShape: bool):
     bpy.context.view_layer.active_layer_collection.collection.objects.link(newPath)
 
     return newPath
+
+
+def updateTiedRoom(scene: Scene, emptyType: str, propName: str, ptrName: str):
+    """Used by handlers to update the tied room object
+
+    Parameters:
+    - ``emptyType``: the OoT empty type name (Water Box, Transition Actor, Entrance, ...)
+    - ``propName``: the properties name to get or set tied room objects
+    - ``ptrName``: the pointer property name
+    """
+
+    if scene.gameEditorMode != "OOT":
+        return
+
+    def updateTiedRoom(parentObj: Object, isScene: bool):
+        for obj in parentObj.children_recursive:
+            if obj.type == "EMPTY":
+                if obj.ootEmptyType == emptyType:
+                    prop = getattr(obj, propName)
+                    tiedRoom = getattr(prop, ptrName)
+                    if not isScene and tiedRoom is None:
+                        setattr(prop, ptrName, parentObj)
+                    elif isScene and tiedRoom is not None:
+                        # case where parent is a scene and the object is a room's child
+                        foundRoom = False
+                        for o in bpy.data.objects:
+                            if o.type == "EMPTY" and o.ootEmptyType == "Room" and obj in o.children_recursive:
+                                foundRoom = True
+                                break
+                        if not foundRoom:
+                            setattr(prop, ptrName, None)
+
+    for parentObj in bpy.data.objects:
+        if parentObj.type == "EMPTY":
+            if parentObj.ootEmptyType == "Room":
+                updateTiedRoom(parentObj, False)
+            elif parentObj.ootEmptyType == "Scene":
+                updateTiedRoom(parentObj, True)
```

Comment:
since you use children_recursive, empties will be processed twice, once per scene and once per room (they're recursively children of both)

depending on who comes first between scene and rooms, the final result will be different, which doesn't sound intended